compiler/test/runnable/test23514.d

```diff
@@ -0,0 +1,9 @@
+// REQUIRED_ARGS: -O
+// https://issues.dlang.org/show_bug.cgi?id=23514
+
+enum offset = 0xFFFF_FFFF_0000_0000UL;
+
+void main()
+{
+    assert((cast(ulong)&main) != (cast(ulong)&main + offset));
```

Comment:
You probably want to check the actual value by taking the address of main as a size_t, then doing the computation on that.